todo-list-backend/src/permissions.ts

```diff
@@ -0,0 +1,35 @@
+import {GraphQLSchema} from "graphql";
+import {allow, rule, shield} from "graphql-shield";
+import {verify} from "jsonwebtoken";
+
+// todo: what is the exact return type of this function? It atleast is a
+//  IMiddlewareGenerator type but the generics are not clear for this one
+export function buildPermissions(schema: GraphQLSchema, tokenSecret: string,
+                                 debugMode: boolean = false): any {
+
+    const isAuthenticated = rule()(async (parent, args, context, info) => {
+        try {
+            let token = verify(context.token, tokenSecret);
```

Comment:
I would *always* verify the token if it's given. Plus, I would always find the currently authenticated user and put that into `context`. That way, you could refactor this rule like that:

```js
const isAuthenticated = rule()(async (parent, args, context, info) => {
  const { user } = context
  return user && user.name
}
```